---
title: Splice ratio comparison for mediator associated and other intron genes.
subtitle: At 30C for all 0h 2h 4h and 6h time points.
date: "`r format(Sys.time(), '%d %B, %Y')`"
author: Mattias Carlsson
output:
  pdf_document:
    latex_engine: xelatex
    toc: true
    number_sections: FALSE
    df_print: kable
mainfont: "Liberation Sans"
monofont: "Liberation Mono"
geometry: a3paper, margin=20mm, landscape
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_file = 
  paste("OutFiles/", format(Sys.time(), '%Y%m%d'),"_",fs::path_ext_remove(basename(inputFile)),".pdf",sep="")
  ) })
---
\newpage

# Intro and method
Requires run of 03_QC_GenerateIntronRatios.R to generate infiles. This script also filter away genes for e.g. low counts which makes the number of genes differ between debatched and non-debatched results.

In debatched results, data was debatched using combat-seq from sva using all 480 samples and two batches defined by rRNA content.

Each data point in box plots is an hexaplicate average.

Batch correction is not expected to have a large effect on results since the same samples are the base for both box plot in each chart.

Interpretation of Results

Splice ratios are compared within each genotype at 30C at different time points.

The splice ratios are not normally distributed but due to large sample sizes t-test is assumed to be acceptable.

p-Values are not corrected for multiple testing but there seems in general to be a difference between mediator and other intron genes at all time points and in all genotypes, but lesser differance in med16 and med18.

At the 6h timepoint uncorrected p-values are highest and for med16 and med18.


\newpage

# 0h timepoint
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_0h.R", local = knitr::knit_global())
```
\newpage

# 0h timepoint debatch
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_0h_Debatch.R", local = knitr::knit_global())
```
\newpage

# 2h timepoint
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_2h.R", local = knitr::knit_global())
```
\newpage

# 2h timepoint debatch
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_2h_Debatch.R", local = knitr::knit_global())
```
\newpage

# 4h timepoint
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_4h.R", local = knitr::knit_global())
```
\newpage

# 4h timepoint debatch
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_4h_Debatch.R", local = knitr::knit_global())
```
\newpage

# 6h timepoint
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_6h.R", local = knitr::knit_global())
```
\newpage

# 6h timepoint debatch
```{r, echo = FALSE, results = 'hide', warning = FALSE, message = FALSE, fig.align='center', out.width = '100%', fig.width = 12, fig.asp = 0.7}
source(file = "06_Compare_Oth-vs-Med_6h_Debatch.R", local = knitr::knit_global())
```

\newpage

# Corrected p-values
```{r, echo = FALSE, warning = FALSE, message = FALSE}
#Generates combination names
GenTimeBatch <- vector(mode = "character")
for(Time in c("0h","2h","4h","6h"))
{
  for(Batch in c("NoDebatch","Debatch"))
  {
    for(Gene in c("Wt","med9","cdk8","med16","med18"))
    {
      GenTimeBatch <- c(GenTimeBatch, paste0(Gene,Time,Batch,collapse = ""))
    }
  }
}
pAdjusted <- p.adjust(pValues, method = "holm")
pValTibble <- tibble(GenTimeBatch,pValues,pAdjusted)
knitr::kable(pValTibble, caption = "From boxplots")
```